### PR TITLE
Fix project references for contracts packages

### PR DESCRIPTION
**Pull request by albrow**
- Removed /packages/contracts
- Removed /contracts/core
- Added `yarn build:ts` to CI tests
- Added each contracts "project" to root tsconfig.json

## Checklist:

*   [ ] Prefix PR title with `[WIP]` if necessary.
*   [ ] Add tests to cover changes as needed.
*   [ ] Update documentation as needed.
*   [ ] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -18,6 +18,7 @@ jobs:
                   name: yarn
                   command: yarn --frozen-lockfile install || yarn --frozen-lockfile install
             - run: yarn build:ci:no_website
+            - run: yarn build:ts
```
Comment (LogvinovLeon) [inline .circleci/config.yml]: Will that build twice on CI and increase the running time of all tests?

Comment (albrow) [reply .circleci/config.yml]: Yes, but only by about 1-2 seconds. The beauty of project references is that `build:ts` is pretty much instant if there are no changes to any TypeScript code 😄

Comment (albrow) [reply .circleci/config.yml]: This is from the latest CI run:

screen shot 2018-12-11 at 4 46 12 pm

**Comment by LogvinovLeon** [comment]
LGMT after tests pass

**Comment by LogvinovLeon** [comment]
Ok, looks like @fabioberger broke them on deveopment

**Comment by albrow** [comment]
Since this PR fixes a bug, @fabioberger suggested merging it. The `test-publish` failures appear unrelated to the contents of this PR and we will need to address it separately.

**Pull request merged**